s2/encode.go

```diff
@@ -366,7 +369,14 @@ const MaxBlockSize = math.MaxUint32 - binary.MaxVarintLen32 - 5
 // 32 bit platforms will have lower thresholds for rejecting big content.
 func MaxEncodedLen(srcLen int) int {
 	n := uint64(srcLen)
-	if n > 0xffffffff {
+	if intReduction == 1 {
```

Comment:
Isn't it more obvious to check whether math/bits.UintSize == 32?

Reply:
Ends up with the same. Potato/Potâto